File: kata/bowling_game/go/2024_02_03/bowling_game.go
package bowling

type Game struct {
	rolls       [21]int
	currentRoll int
}

func NewGame() *Game {
	return &Game{
		currentRoll: 0,
	}
}

func (g *Game) Roll(pins int) {
	g.rolls[g.currentRoll] = pins
	g.currentRoll++
}

func (g *Game) Score() int {
	score := 0
	frameIndex := 0
	for frame := 0; frame < 10; frame++ {
		if g.isStrile(frameIndex) {
			score += 10 + g.rolls[frameIndex+1] + g.rolls[frameIndex+2]
			frameIndex += 1
		} else if g.isSpare(frameIndex) {
			score += 10 + g.rolls[frameIndex+2]
			frameIndex += 2
		} else {
			score += g.rolls[frameIndex] + g.rolls[frameIndex+1]
			frameIndex += 2
		}
	}

	return score
}

func (g *Game) isStrile(frameIndex int) bool {
	return g.rolls[frameIndex] == 10
}
func (g *Game) isSpare(frameIndex int) bool {
	return g.rolls[frameIndex]+g.rolls[frameIndex+1] == 10
}

File: kata/bowling_game/python/2025_06_26/bowling_game.py
class Game:
    def __init__(self):
        self.rolls = []

    def roll(self, pins):
        self.rolls.append(pins)

    def score(self):
        score = 0
        frame_index = 0

        for frame in range(10):
            if self.rolls[frame_index] + self.rolls[frame_index + 1] == 10:
                score += 10 + self.rolls[frame_index + 2]
                frame_index += 2
            else:
                score += self.rolls[frame_index] + self.rolls[frame_index + 1]
                frame_index += 2

        return score
